File: EcoBookAiBackend/src/main/java/com/ecobook/dto/UserNotificationDTO.java
package com.ecobook.dto;

import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;
import java.util.Map;

@Getter
@Builder
public class UserNotificationDTO {

    private final String id;
    private final String title;
    private final String body;
    private final String notificationType;
    private final String route;
    private final String requestId;
    private final String materialId;
    private final LocalDateTime receivedAt;
    private final boolean unread;
    @Builder.Default
    private final Map<String, String> metadata = Map.of();
}

File: EcoBookAiAndroid/app/src/main/kotlin/com/ecobook/dto/UserNotificationDTO.kt
package com.ecobook.dto

data class UserNotificationDTO(
    val id: String? = null,
    val title: String? = null,
    val body: String? = null,
    val notificationType: String? = null,
    val route: String? = null,
    val requestId: String? = null,
    val materialId: String? = null,
    val receivedAt: String? = null,
    val unread: Boolean? = null,
    val metadata: Map<String, String> = emptyMap()
)
